compiler/src/dotty/tools/dotc/transform/OverridingPairs.scala

```diff
@@ -40,7 +40,7 @@ object OverridingPairs {
      *  relative to <base>.this do
      */
     protected def matches(sym1: Symbol, sym2: Symbol): Boolean =
-      sym1.isType || self.memberInfo(sym1).matches(self.memberInfo(sym2))
+      sym1.isType || (!exclude(sym2) && self.memberInfo(sym1).matches(self.memberInfo(sym2)))
```

Comment:
Why is this change needed ? Doesn't the existing logic already take care of calling exclude when necessary ?

Reply:
You're right, seems to be working without this change